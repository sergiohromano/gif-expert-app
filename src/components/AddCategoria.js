import React, { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategoria = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState("");

    const handleSubmit = (e) => {
        if (inputValue.trim().length > 2) {
            e.preventDefault();
            setCategories((cats) => [inputValue, ...cats]);
            setInputValue("");
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    );
}

AddCategoria.propTypes = {
    setCategories: PropTypes.func.isRequired,
}