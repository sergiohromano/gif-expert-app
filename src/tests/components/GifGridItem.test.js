import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />',() => {
    const title = 'Titulo de prueba';
    const url = 'https://localhost/imagen.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe de mostrar el component correctamente',() => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un pÃ¡rrafo con el title',() => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props',() => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe de tener la clase animate__animated',() => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__animated')).toBe(true);
    })
})