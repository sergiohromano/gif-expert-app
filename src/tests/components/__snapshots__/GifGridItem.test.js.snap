// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el component correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de prueba"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
